hadoop-hdds/common/src/test/java/org/apache/hadoop/utils/db/TestDBStoreBuilder.java

```diff
@@ -71,12 +71,26 @@ public void builderWithOneParamV2() throws IOException {
     if(!newFolder.exists()) {
       Assert.assertTrue(newFolder.mkdirs());
     }
-    thrown.expect(IOException.class);
     DBStoreBuilder.newBuilder(conf)
         .setPath(newFolder.toPath())
         .build();
   }
 
+  @Test
+  public void builderWithWalDirSet() throws IOException {
+    Configuration conf = new Configuration();
+    File newFolder = folder.newFolder();
+    File walDir = folder.newFolder();
+    if(!newFolder.exists()) {
+      Assert.assertTrue(newFolder.mkdirs());
+    }
+    DBStoreBuilder.newBuilder(conf)
+        .setPath(newFolder.toPath())
```

Comment:
This test illustrates my concern. We will add this feature for performance reasons but we really have no data to show or prove this is the root cause of perf issues. Neither can we test it.